bin/varnishd/cache/cache_esi_fetch.c

```diff
@@ -165,8 +165,13 @@ vfp_esi_gzip_init(struct vfp_ctx *vc, struct vfp_entry *vfe)
 	if (vef == NULL)
 		return (VFP_ERROR);
 	vc->obj_flags |= OF_GZIPED | OF_CHGCE | OF_ESIPROC;
-	vef->vgz = VGZ_NewGzip(vc->wrk->vsl, "G F E");
 	vef->vep = VEP_Init(vc, vc->req, vfp_vep_callback, vef);
+	if (vef->vep == NULL) {
+		FREE_OBJ(vef);
+		return (VFP_ERROR);
+	}
+	vef->vgz = VGZ_NewGzip(vc->wrk->vsl, "G F E");
```

Comment:
I'll add an `AN()`.

In case you meant to imply that we should actually _handle_ a failed vgz init: I think it is fair to say that, in general, we do not handle `malloc()` failures (mostly via `ALLOC_OBJ()`). If we _actually_ wanted to manage low memory conditions, I think we should first make that general decision and then systematically work all the places. And I do not want to think of the substantial testing effort.

Reply:
I think we could turn error handling into an assertion in `vfp_gzip_init()` to be comprehensive, this is probably a dead branch:

https://github.com/varnishcache/varnish-cache/blob/12df4f60af2ce3cebd26b0075d98908a21109785/bin/varnishd/cache/cache_gzip.c#L506-L507

Reply:
right. force pushed